app/src/main/java/com/amaze/filemanager/database/UtilsHandler.java

```diff
@@ -113,7 +117,55 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+                        "INSERT INTO " + backupTable + " SELECT * FROM " + TABLE_HISTORY + " group by path;" +
+                        "DROP TABLE " + TABLE_HISTORY + ";" +
+                        "ALTER TABLE " + backupTable + " RENAME TO " + TABLE_HISTORY + ";";
+                db.execSQL(backupHistory);
+
+                backupTable = "temp_" + TABLE_HIDDEN;
+                String backupHidden = queryHistory.replace(TABLE_HIDDEN, backupTable) +
+                        "INSERT INTO " + backupTable + " SELECT * FROM " + TABLE_HIDDEN + " group by path;" +
+                        "DROP TABLE " + TABLE_HIDDEN + ";" +
+                        "ALTER TABLE " + backupTable + " RENAME TO " + TABLE_HIDDEN + ";";
+                db.execSQL(backupHidden);
+
+                backupTable = "temp_" + TABLE_LIST;
+                String backupList = queryHistory.replace(TABLE_LIST, backupTable) +
+                        "INSERT INTO " + backupTable + " SELECT * FROM " + TABLE_LIST + " group by path;" +
+                        "DROP TABLE " + TABLE_LIST + ";" +
+                        "ALTER TABLE " + backupTable + " RENAME TO " + TABLE_LIST + ";";
+                db.execSQL(backupList);
+
+                backupTable = "temp_" + TABLE_GRID;
+                String backupGrid = queryHistory.replace(TABLE_GRID, backupTable) +
+                        "INSERT INTO " + backupTable + " SELECT * FROM " + TABLE_GRID + " group by path;" +
+                        "DROP TABLE " + TABLE_GRID + ";" +
+                        "ALTER TABLE " + backupTable + " RENAME TO " + TABLE_GRID + ";";
+                db.execSQL(backupGrid);
+
+                backupTable = "temp_" + TABLE_BOOKMARKS;
+                String backupBookmarks = queryHistory.replace(TABLE_BOOKMARKS, backupTable) +
+                        "INSERT INTO " + backupTable + " SELECT * FROM " + TABLE_BOOKMARKS + " group by path;" +
+                        "DROP TABLE " + TABLE_BOOKMARKS + ";" +
+                        "ALTER TABLE " + backupTable + " RENAME TO " + TABLE_BOOKMARKS + ";";
+                db.execSQL(backupBookmarks);
+
+                backupTable = "temp_" + TABLE_SMB;
+                String backupSmbs = queryHistory.replace(TABLE_SMB, backupTable) +
+                        "INSERT INTO " + backupTable + " SELECT * FROM " + TABLE_SMB + " group by path;" +
+                        "DROP TABLE " + TABLE_SMB + ";" +
+                        "ALTER TABLE " + backupTable + " RENAME TO " + TABLE_SMB + ";";
+                db.execSQL(backupSmbs);
+
+                backupTable = "temp_" + TABLE_SFTP;
+                String backup = queryHistory.replace(TABLE_SFTP, backupTable) +
+                        "INSERT INTO " + backupTable + " SELECT * FROM " + TABLE_SFTP + " group by path;" +
+                        "DROP TABLE " + TABLE_SFTP + ";" +
+                        "ALTER TABLE " + backupTable + " RENAME TO " + TABLE_SFTP + ";";
+                db.execSQL(backup);
```

Comment:
Tested on Oneplus 2 running Slim7 (7.1.2).

I made a little tweak to log the SQL statements ran from above, and I got:
```
09-27 15:41:32.800 21670-21693/com.amaze.filemanager D/DEBUG_SQL: CREATE TABLE IF NOT EXISTS temp_history (_id INTEGER PRIMARY KEY,path TEXT UNIQUE);INSERT INTO temp_history SELECT * FROM history group by path;DROP TABLE history;ALTER TABLE temp_history RENAME TO history;
09-27 15:41:32.802 21670-21693/com.amaze.filemanager D/DEBUG_SQL: CREATE TABLE IF NOT EXISTS history (_id INTEGER PRIMARY KEY,path TEXT UNIQUE);INSERT INTO temp_hidden SELECT * FROM hidden group by path;DROP TABLE hidden;ALTER TABLE temp_hidden RENAME TO hidden;
09-27 15:41:32.803 21670-21693/com.amaze.filemanager D/DEBUG_SQL: CREATE TABLE IF NOT EXISTS history (_id INTEGER PRIMARY KEY,path TEXT UNIQUE);INSERT INTO temp_list SELECT * FROM list group by path;DROP TABLE list;ALTER TABLE temp_list RENAME TO list;
    CREATE TABLE IF NOT EXISTS history (_id INTEGER PRIMARY KEY,path TEXT UNIQUE);INSERT INTO temp_grid SELECT * FROM grid group by path;DROP TABLE grid;ALTER TABLE temp_grid RENAME TO grid;
09-27 15:41:32.804 21670-21693/com.amaze.filemanager D/DEBUG_SQL: CREATE TABLE IF NOT EXISTS history (_id INTEGER PRIMARY KEY,path TEXT UNIQUE);INSERT INTO temp_bookmarks SELECT * FROM bookmarks group by path;DROP TABLE bookmarks;ALTER TABLE temp_bookmarks RENAME TO bookmarks;
09-27 15:41:32.805 21670-21693/com.amaze.filemanager D/DEBUG_SQL: CREATE TABLE IF NOT EXISTS history (_id INTEGER PRIMARY KEY,path TEXT UNIQUE);INSERT INTO temp_smb SELECT * FROM smb group by path;DROP TABLE smb;ALTER TABLE temp_smb RENAME TO smb;
09-27 15:41:32.806 21670-21693/com.amaze.filemanager D/DEBUG_SQL: CREATE TABLE IF NOT EXISTS history (_id INTEGER PRIMARY KEY,path TEXT UNIQUE);INSERT INTO temp_sftp SELECT * FROM sftp group by path;DROP TABLE sftp;ALTER TABLE temp_sftp RENAME TO sftp;
```

The `history` table was created multiple times, which means your `String.replace(old, new)` didn't work as you expect.

For example, at line 129 your `queryHistory.replace()` was trying to replace `hidden` with `temp_hidden` and `hidden` never existed in `queryHistory`.

Also, (I think it's a consequence from above mistake) `temp_history` table didn't renamed to `history` after running the upgrade SQL statements. Please check.

Reply:
Yes, I copy code and forget change `queryHistory`. I have updated.

Reply:
Now I can see the SQLs are generated correctly. However the `temp_` tables are not populated, nor the `temp_` tables were not renamed and the original tables were not dropped.

Perhaps Android/sqlite APIs doesn't support running multiple SQL statements in a single `String`? Maybe try running the 3 statements per table one by one?

Tested with Oneplus 2 running Slim7 (7.1.2).